### DISCUSSION TITLE
Does MS Teams have documentation for popup windows?

### DISCUSSION THREAD

**Discussion by BustosAndrew** [pnp/generator-teams#204] [Q&A] [answered]
I seem to not be able to find documentation for creating a pop up window to authenticate a user for the first time. How would we go about creating a pop up window during runtime in MS Teams if anyone knows?

**Comment by wictorwilen** [comment, marked as answer]
It's called **Task Modules** in Teams. See https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/what-are-task-modules

**Reply by BustosAndrew** [reply]
Well I hope it's something that can allow me to pass an authentication code to the main tab.

**Reply by wictorwilen** [reply]
I missed the *auth* part of the question - yes, for auth scenarios use the `microsoftTeams.authentication.authenticate()` method as described in: https://developer.microsoft.com/en-us/microsoft-teams/blogs/authentication-in-microsoft-teams-apps-tabs/#:~:text=In%20the%20tab%20content%20or%20configuration%20page%2C%20call,a%20popup%20that%20will%20host%20the%20authentication%20flow.

**Reply by BustosAndrew** [reply]
Oh I meant to pass in an access code from a different api. Box's API to be specific. I think the task window can probably help me with getting the access code into the tab's content page.
